src/Workspaces/Core/MSBuild/Microsoft.CodeAnalysis.Workspaces.MSBuild.csproj

```diff
@@ -5,7 +5,7 @@
     <OutputType>Library</OutputType>
     <RootNamespace>Microsoft.CodeAnalysis</RootNamespace>
     <TargetFrameworks>$(SourceBuildTargetFrameworks);net472</TargetFrameworks>
-    <DefineConstants>$(DefineConstants);WORKSPACE_MSBUILD</DefineConstants>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```

Comment:
why?

Reply:
Some of the adding-and-removing references in the project file logic calls into methods in the CLR to ask where the GAC path is which uses unsafe code. You'll see the AllowUnsafeBlocks being removed from the BuildHost project since that is just code moving.